tests/framework/utils.py

```diff
@@ -193,8 +193,8 @@ def with_arg(self, flag, value=""):
     def build(self):
         """Build the command."""
         cmd = self._bin_path + " "
-        for flag in self._args:
-            cmd += flag + " " + "{}".format(self._args[flag]) + " "
+        for (flag, value) in self._args.items():
+            cmd += flag + " " + "{}".format(value) + " "
```

Comment:
Would look nicer with `f-strings`.

Reply:
yes, most of the `.format`'s should be replaced by f-strings, but it's a titanic amount of work to do this :)

Reply:
Changed this specific one, since it indeed looks guite ugly